metals/src/main/scala/scala/meta/internal/metals/SemanticTokensProvider.scala

```diff
@@ -170,69 +174,80 @@ object SemanticTokensProvider {
+        case _: Token.KwFalse => getTypeId(SemanticTokenTypes.Keyword)
 
-      // Constant
-      case _: Token.Constant.Int | _: Token.Constant.Long |
-          _: Token.Constant.Float | _: Token.Constant.Double =>
-        getTypeId(SemanticTokenTypes.Number)
-      case _: Token.Constant.String | _: Token.Constant.Char =>
-        getTypeId(SemanticTokenTypes.String)
-      case _: Token.Constant.Symbol => getTypeId(SemanticTokenTypes.Property)
+        // extends Symbolic keywords
+        case _: Token.Hash => getTypeId(SemanticTokenTypes.Keyword)
+        case _: Token.Viewbound => getTypeId(SemanticTokenTypes.Operator)
+        case _: Token.LeftArrow => getTypeId(SemanticTokenTypes.Operator)
+        case _: Token.Subtype => getTypeId(SemanticTokenTypes.Keyword)
+        case _: Token.RightArrow => getTypeId(SemanticTokenTypes.Operator)
+        case _: Token.Supertype => getTypeId(SemanticTokenTypes.Keyword)
+        case _: Token.At => getTypeId(SemanticTokenTypes.Keyword)
+        case _: Token.Underscore => getTypeId(SemanticTokenTypes.Variable)
+        case _: Token.TypeLambdaArrow => getTypeId(SemanticTokenTypes.Operator)
+        case _: Token.ContextArrow => getTypeId(SemanticTokenTypes.Operator)
 
-      // Comment
-      case _: Token.Comment => getTypeId(SemanticTokenTypes.Comment)
+        // Constant
+        case _: Token.Constant.Int | _: Token.Constant.Long |
+            _: Token.Constant.Float | _: Token.Constant.Double =>
+          getTypeId(SemanticTokenTypes.Number)
+        case _: Token.Constant.String | _: Token.Constant.Char =>
+          getTypeId(SemanticTokenTypes.String)
+        case _: Token.Constant.Symbol => getTypeId(SemanticTokenTypes.Property)
 
-      // Interpolation
-      case _: Token.Interpolation.Id | _: Token.Interpolation.SpliceStart =>
-        getTypeId(SemanticTokenTypes.Keyword)
-      case _: Token.Interpolation.Start | _: Token.Interpolation.Part |
-          _: Token.Interpolation.SpliceEnd | _: Token.Interpolation.End =>
-        getTypeId(SemanticTokenTypes.String) // $ symbol
+        // Comment
+        case _: Token.Comment => getTypeId(SemanticTokenTypes.Comment)
 
-      case _ if isScala3 =>
-        trySoftKeyword(tk)
+        // Interpolation
+        case _: Token.Interpolation.Id | _: Token.Interpolation.SpliceStart =>
+          getTypeId(SemanticTokenTypes.Keyword)
+        case _: Token.Interpolation.Start | _: Token.Interpolation.Part |
+            _: Token.Interpolation.SpliceEnd | _: Token.Interpolation.End =>
+          getTypeId(SemanticTokenTypes.String) // $ symbol
 
-      // Default
-      case _ => tokenFallback(tk)
-    }
+        case _ if isScala3 && trySoftKeyword(tk) != -1 =>
+          trySoftKeyword(tk)
+        case _ => -1
+      }
+
+    val tokenModifier: Int =
+      tokenType match {
+        case _ if tokenType == getTypeId(SemanticTokenTypes.Variable) =>
+          1 << getModifierId(SemanticTokenModifiers.Readonly)
```

Comment:
Shouldn't we add readonly only in the fallback?

Reply:
This is for cases when token isn't an ident, but it becomes variable. For now, the only case for that is underscore, which is almost always readonly.